### PR TITLE
feat: application role connections

### PR DESCRIPTION
**Pull request by plun1331**
## Summary

https://github.com/discord/discord-api-docs/pull/5668

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

File: discord/application_role_connection.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-2021 Rapptz
```
Comment (Dorukyum) [inline discord/application_role_connection.py:4]: New files are only owned by Pycord Development.
```suggestion
```

File: discord/types/application_role_connection.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2015-2021 Rapptz
```
Comment (Dorukyum) [inline discord/types/application_role_connection.py:4]:
```suggestion
```

**Review by Dorukyum — DISMISSED** [review]

**Review by Lulalaby — APPROVED** [review]

**Review by Lulalaby — DISMISSED** [review]
I hate github sometimes

File: discord/application_role_connection.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from .enums import ApplicationRoleConnectionMetadataType, try_enum
+from .utils import MISSING
+
+if TYPE_CHECKING:
+    from .state import ConnectionState
+    from .types.guild import (
```
Comment (EmreTech) [inline discord/application_role_connection.py:34]: `ApplicationRoleConnection` isn't defined in `discord.types.guild`?

File: discord/application_role_connection.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        ApplicationRoleConnection as ApplicationRoleConnectionPayload,
+    )
+
+
+class ApplicationRoleConnection:
```
Comment (EmreTech) [inline discord/application_role_connection.py:39]: Docstring.

File: discord/http.py [hunk 5]
```diff
@@ -2817,6 +2818,31 @@ def bulk_edit_guild_application_command_permissions(
...
+            "GET",
+            "/applications/{application_id}/role-connections/metadata",
+            application_id=application_id,
+        )
+        return self.request(r)
+
+    def update_application_role_connection_metadata_records(
+        self,
+        application_id: Snowflake,
+        payload,  # TODO: payload typehint, check return type
```
Comment (EmreTech) [inline discord/http.py:2837]: The payload has been defined, so it's about time to type hint this parameter.

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: discord/application_role_connection.py (hunk 3 above)
Comment (Lulalaby) [reply discord/application_role_connection.py:34]: mhh thought about this rn.
role connection isn't a guild type in this context at all.

Comment (plun1331) [reply discord/application_role_connection.py:34]: Oopsie

**Review by BobDotCom — DISMISSED** [review]
Please add a changelog entry

**Comment by Lulalaby** [comment]
State update: In rollout

**Review by Lulalaby — APPROVED** [review]

**Comment by plun1331** [comment]
Haven't actually tested yet

**Review by Nzii3 — APPROVED** [review]

**Comment by BobDotCom** [comment]
@plun1331 please resolve merge conflicts

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,13 +10,17 @@ possible (see our [Version Guarantees] for more info).
 
 These changes are available on the `master` branch, but have not yet been released.
 
-### Added
+## Added
```
Comment (Lulalaby) [inline CHANGELOG.md:13]:
```suggestion
### Added
```

God damn it

Comment (JustaSqu1d) [reply CHANGELOG.md]: `##` Indicates the version.
`###` Indicates the type of change made.

Comment (Lulalaby) [reply CHANGELOG.md]: Added is not a version ;)

**Comment by BobDotCom** [comment]
A thought: Maybe we remove "Application" from the naming to shorten it? `ApplicationRoleConnectionMetadataPayload` is a *really* long class name.

**Comment by Lulalaby** [comment]
> A thought: Maybe we remove "Application" from the naming to shorten it? `ApplicationRoleConnectionMetadataPayload` is a _really_ long class name.

phew, not a fan, because it's an application role connection and not a normal role connection

**Comment by BobDotCom** [comment]
Status update?

**Comment by DTheIcyDragon** [comment]
any updates on this? this feature is meanwhile rolling out to every guild

**Comment by Lulalaby** [comment]
> any updates on this? this feature is meanwhile rolling out to every guild

We are aware

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**